solidity/contracts/libs/isms/MultisigIsmMetadata.sol

```diff
@@ -3,40 +3,35 @@ pragma solidity >=0.8.0;
 
 /**
  * Format of metadata:
- * [   0:  32] Merkle root
- * [  32:  36] Root index
- * [  36:  68] Origin mailbox address
- * [  68:1092] Merkle proof
- * [1092:????] Validator signatures, 65 bytes each, length == Threshold
+ * [   0:  32] Origin mailbox address
+ * [  32:  VS] Validator signatures (where VS := 32 + Threshold * 65 bytes)
+ * [  VS:????] Suffix
+ *
+ * Format of suffix:
+ * [  -38:  -1] Merkle root
```

Comment:
I don't understand these negative indices, definitely warrant a comment.

Consider using positive indices and making it clear that they are relative to the suffix start